src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -214,6 +214,7 @@ trait ScalaSettings extends AbsScalaSettings
   val stopAfter       = PhasesSetting     ("-Ystop-after", "Stop after") withAbbreviation ("-stop") // backward compat
   val stopBefore      = PhasesSetting     ("-Ystop-before", "Stop before")
   val Yrangepos       = BooleanSetting    ("-Yrangepos", "Use range positions for syntax trees.")
+  val Yvalidatepos    = PhasesSetting     ("-Yvalidate-pos", "Validate positions after the given phases") withPostSetHook (_ => Yrangepos.value = true)
```

Comment:
Enabling range pos automatically is might come as a surprise. Let's mention (" (enables -Yrangepos)" in the help for this and `-Ymemberpos`.

Reply:
We should turn this on by default in `interactive.Global`, ala https://github.com/scala/scala/blob/d03d6a92d4c50cf834bcf5b22d093cc62176003f/src/interactive/scala/tools/nsc/interactive/Global.scala#L119-L120